src/passes/OptimizeInstructions.cpp

```diff
@@ -895,9 +895,13 @@ struct OptimizeInstructions
   // Canonicalizing the order of a symmetric binary helps us
   // write more concise pattern matching code elsewhere.
   void canonicalize(Binary* binary) {
-    assert(isSymmetric(binary));
+    assert(isSymmetricOrRelational(binary));
+    bool isRelational = binary->isRelational();
     auto swap = [&]() {
       assert(canReorder(binary->left, binary->right));
+      if (isRelational) {
```

Comment:
```suggestion
      if (binary->isRelational()) {
```
This will let you avoid needing a new variable.